### DISCUSSION TITLE
Can we get a newer version of tiangolo/uvicorn-gunicorn-fastapi with python3.12 and up-to-date packages?

### DISCUSSION THREAD

**Discussion by Pyroseza** [tiangolo/uvicorn-gunicorn-docker#224] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched in the documentation/README.
- [X] I already searched in Google "How to do X" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs/README and didn't find an answer.

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```Text
I would like to be able to add this to my docker file and have it work:


FROM tiangolo/uvicorn-gunicorn-fastapi:python3.12
```
```


### Description

This request is to get a newer version of the uvicorn-gunicorn-fastapi image with Python 3.12 (3.13 is coming out soon) and to have all the dependencies updated.

I ran a test container with `FROM tiangolo/uvicorn-gunicorn-fastapi:python3.11` and this is what the installed packages are, take note that **FastAPI itself it 0.88.0 which is many many versions behind the latest development!!!**
```
# pip freeze
anyio==4.4.0
certifi==2024.7.4
click==8.1.7
dnspython==2.6.1
email_validator==2.2.0
fastapi==0.88.0
gunicorn==22.0.0
h11==0.14.0
httpcore==1.0.5
httptools==0.6.1
httpx==0.27.0
idna==3.7
itsdangerous==2.2.0
Jinja2==3.1.4
MarkupSafe==2.1.5
orjson==3.10.7
packaging==24.1
pydantic==1.10.17
python-dotenv==1.0.1
python-multipart==0.0.9
PyYAML==6.0.1
sniffio==1.3.1
starlette==0.22.0
typing_extensions==4.12.2
ujson==5.10.0
uvicorn==0.20.0
uvloop==0.19.0
watchfiles==0.22.0
websockets==12.0
```

### Operating System

Linux

### Operating System Details

I'm running this on Ubuntu

### Python Version

3.12

### Additional Context

_No response_

**Comment by tiangolo** [comment, marked as answer]
Now that Uvicorn supports managing workers with `--workers`, including restarting dead ones, there's no need for Gunicorn. That also means that it's much simpler to build a Docker image from scratch now, I updated the docs to explain it.

Because of that, I deprecated this Docker image: https://github.com/tiangolo/uvicorn-gunicorn-docker#-warning-you-probably-dont-need-this-docker-image
